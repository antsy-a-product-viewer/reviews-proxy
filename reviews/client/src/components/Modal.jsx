import React from 'react';
import moment from 'moment';
import _ from 'underscore';
import Star from './Star.jsx';
import Item from './Item.jsx';
import modalStyles from './css/modalStyles.css.js';

const Modal = (props) => {
  if (!props.showModal) {
    return null;
  } else {
    return (
      <div className="modal" id="modal-overlay" style={modalStyles.modal} onClick={props.onClose}>
        <section style={modalStyles.modalMain}>
          <div>
            <img style={modalStyles.image} src={props.review.review_img}></img>
          </div>
          <div style={{display: 'flex', flexDirection: 'column', justifyContent: 'space-between'}}>
            <div className="modalReview" style={{marginTop: 20}}>
              <div style={modalStyles.userInfo}>
                <img style ={modalStyles.profileImg} src={props.review.user_img}></img>
                <div style={modalStyles.username}>{props.review.user_name}</div>
                <div style={modalStyles.date}>{moment(props.review.created_at).format('MMM DD, YYYY')}</div>
              </div>
              <div style={modalStyles.review}>
                <Star stars={props.review.stars}/>
                <div>{props.review.review}</div>
              </div>
            </div>
            <div className="modalItem">
              <div style={modalStyles.item}>
                <Item showPrice="true" review={props.review} visitItem={props.visitItem}/>
              </div>
            </div>
          </div>
          <button className="closeModal"style={modalStyles.button} onClick={props.onClose}>
            <img id="close-modal" style={modalStyles.buttonImg} src="https://cdn0.iconfinder.com/data/icons/elasto-online-store/26/00-ELASTOFONT-STORE-READY_close-512.png"></img>
          </button>
        </section>
      </div>
    );
  }
};

export default Modal;
