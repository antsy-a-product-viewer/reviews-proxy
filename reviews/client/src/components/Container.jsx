import React from 'react';
import axios from 'axios';
import _ from 'underscore';
import Star from './Star.jsx';
import ReviewPhotosContainer from './ReviewPhotosContainer.jsx';
import Modal from './Modal.jsx';
import containerStyles from './css/containerStyles.css.js';
import ReviewContainer from './ReviewContainer.jsx';
import Button from './Button.jsx';

class Container extends React.Component {
  constructor(props) {
    super(props);
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.getData = this.getData.bind(this);
    this.getReviews = this.getReviews.bind(this);
    this.getReviewImages = this.getReviewImages.bind(this);
    this.getAverageStars = this.getAverageStars.bind(this);
    this.showMore = this.showMore.bind(this);
    this.visitItem = this.visitItem.bind(this);
    this.reviewImageClick = this.reviewImageClick.bind(this);
    this.state = {
      storeId: 0,
      averageStars: 0,
      reviews: [],
      reviewImages: [],
      showModal: false,
      show: 4,
      modalReview: null
    };
  }

  componentDidMount() {
    this.getData();
  }

  openModal(reviewImgIndex) {
    this.setState({
      modalReview: reviewImgIndex,
      showModal: !this.state.showModal
    });
  }

  closeModal(e) {
    if (e.target.id === 'modal-overlay' || e.target.id === 'close-modal') {
      this.setState({
        showModal: !this.state.showModal,
        modalReview: null
      });
    }
  }

  getData() {
    var endpoint = window.location.pathname + 'store_id';
    axios.get(endpoint)
      .then((res) => {
        var storeId = res.data[0].store_id;
        this.setState({
          storeId: storeId
        });
        this.getReviews(storeId);
        this.getReviewImages(storeId);
      })
      .catch((err) => {
        console.log(err);
      });
  }

  getReviews(storeId) {
    axios.get(`/stores/${storeId}/reviews`)
      .then((res) => {
        this.getAverageStars(res.data);
        this.setState({
          reviews: res.data
        });
      })
      .catch((err) => {
        console.log(err);
      }); 
  }

  getReviewImages(storeId) {
    axios.get(`/stores/${storeId}/review_images`)
      .then((res) => {
        this.setState({
          reviewImages: res.data
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  getAverageStars(reviews) {
    var total = 0;
    for (var i = 0; i < reviews.length; i++) {
      total += reviews[i].stars;
    }
    var average = Math.floor(total / reviews.length);
    this.setState({
      averageStars: average
    });
  }

  showMore() {
    this.setState({
      show: 20
    });
  }
  
  visitItem(productId) {
    window.location.pathname = `/product/${productId}`;
  }

  reviewImageClick(id) {
    this.setState({
      modalReview: _.findIndex(this.state.reviewImages, {review_id: parseInt(id)}),
      showModal: true
    });
  }
  
  render() {
    return (
      <div style={containerStyles.container}>
        <div style={{display: 'flex', marginBottom: '-10px'}}>
          <div style={containerStyles.header}>
            Reviews
          </div>
          <Star starClass="averageStars" stars={this.state.averageStars}/>
          <div style={containerStyles.centeredDiv}>
            ({this.state.reviews.length})
          </div>
        </div>
        <ReviewContainer reviews={this.state.reviews} limit={this.state.show} showPrice="false" visitItem={this.visitItem} imageClick={this.reviewImageClick}/>
        <Button currentNumber={this.state.show} showMore={this.showMore} totalReviews={this.state.reviews.length}/>
        <ReviewPhotosContainer reviewImages={this.state.reviewImages} openModal={this.openModal}/>
        <Modal showModal={this.state.showModal} onClose={this.closeModal} visitItem={this.visitItem} review={this.state.reviewImages[this.state.modalReview]}/>
      </div>
    );
  }
}


export default Container;