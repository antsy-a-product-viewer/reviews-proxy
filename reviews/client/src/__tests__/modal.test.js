import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import Modal from '../components/Modal.jsx';
import Container from '../components/Container.jsx'

configure({ adapter: new Adapter() });

it('should render correctly when props.showModal is true', () => {
  const props = {
    modalReview: 0,
    showModal: true
  };
  const modal = shallow(<Modal showModal={props.showModal} review={reviewImages[props.modalReview]}/>);
  
  expect(modal).toMatchSnapshot();
});

// it('should display the modal when showModal state is true', () =>{
//   const props = {
//     modalReview: 0,
//     showModal: true
//   };
//   const modal = mount(<Modal showModal={props.showModal} review={reviewImages[props.modalReview]}/>)

//   console.log(container.find('.modal').debug());

//   expect(container.find('.modal').exists()).toEqual(true);
// });

const reviewImages = [
  {
    "review_id": 242,
    "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00242.jpg",
    "review": "You can't calculate the panel without bypassing the mobile SMS array!",
    "stars": 2,
    "created_at": "2014-08-02T02:54:51.000Z",
    "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00087.jpg",
    "item_name": "Rustic Plastic Chips",
    "price": 474,
    "user_name": "Peter Blick DDS",
    "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00016.jpg"
  }
];
