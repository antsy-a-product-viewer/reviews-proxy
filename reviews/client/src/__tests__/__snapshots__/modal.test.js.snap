// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when props.showModal is true 1`] = `
<div
  className="modal"
  id="modal-overlay"
  style={
    Object {
      "alignItems": "center",
      "animation": "show .5s ease",
      "backgroundColor": "rgba(0, 0, 0, 0.7)",
      "bottom": 0,
      "display": "flex",
      "justifyContent": "center",
      "left": 0,
      "opacity": 1,
      "overflowX": "hidden",
      "overflowY": "auto",
      "padding": "1rem",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "zIndex": "9999",
    }
  }
>
  <section
    style={
      Object {
        "background": "white",
        "display": "flex",
        "height": 565,
        "left": "50%",
        "position": "fixed",
        "top": "45%",
        "transform": "translate(-50%,-50%)",
        "width": "65%",
      }
    }
  >
    <div>
      <img
        src="https://s3-us-west-1.amazonaws.com/antsyreviewimages/00242.jpg"
        style={
          Object {
            "height": 565,
            "width": 565,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        className="modalReview"
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "margin": 20,
            }
          }
        >
          <img
            src="https://s3-us-west-1.amazonaws.com/antsyuserimages/00016.jpg"
            style={
              Object {
                "borderRadius": 50,
                "height": 50,
                "width": 50,
              }
            }
          />
          <div
            style={
              Object {
                "alignSelf": "center",
                "margin": 5,
                "marginLeft": 15,
                "textDecoration": "underline",
              }
            }
          >
            Peter Blick DDS
          </div>
          <div
            style={
              Object {
                "alignSelf": "center",
                "margin": 5,
              }
            }
          >
            Aug 01, 2014
          </div>
        </div>
        <div
          style={
            Object {
              "margin": 20,
            }
          }
        >
          <Star
            stars={2}
          />
          <div>
            You can't calculate the panel without bypassing the mobile SMS array!
          </div>
        </div>
      </div>
      <div
        className="modalItem"
      >
        <div
          style={
            Object {
              "margin": 20,
            }
          }
        >
          <Item
            review={
              Object {
                "created_at": "2014-08-02T02:54:51.000Z",
                "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00087.jpg",
                "item_name": "Rustic Plastic Chips",
                "price": 474,
                "review": "You can't calculate the panel without bypassing the mobile SMS array!",
                "review_id": 242,
                "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00242.jpg",
                "stars": 2,
                "user_img": "https://s3-us-west-1.amazonaws.com/antsyuserimages/00016.jpg",
                "user_name": "Peter Blick DDS",
              }
            }
            showPrice="true"
          />
        </div>
      </div>
    </div>
    <button
      className="closeModal"
      style={
        Object {
          "background": "none",
          "border": "none",
          "cursor": "pointer",
          "display": "block",
          "float": "right",
          "padding": "none",
          "position": "absolute",
          "right": 5,
          "top": 5,
          "zIndex": 3,
        }
      }
    >
      <img
        id="close-modal"
        src="https://cdn0.iconfinder.com/data/icons/elasto-online-store/26/00-ELASTOFONT-STORE-READY_close-512.png"
        style={
          Object {
            "background": "white",
            "height": 20,
            "margin": 10,
            "width": 20,
          }
        }
      />
    </button>
  </section>
</div>
`;
